cpp/oneapi/dal/algo/kmeans/test/batch.cpp

```diff
@@ -0,0 +1,427 @@
+        SECTION("label shape is expected") {
+            REQUIRE(left.get_row_count() == right.get_row_count());
+            REQUIRE(left.get_column_count() == right.get_column_count());
+            REQUIRE(left.get_column_count() == 1);
+        }
+        SECTION("label match is expected") {
+            const auto left_rows = row_accessor<const Float>(left).pull({ 0, -1 });
+            const auto right_rows = row_accessor<const Float>(right).pull({ 0, -1 });
+            bool failed = false;
+            for (std::int64_t i = 0; i < left_rows.get_count(); i++) {
+                const Float l = left_rows[i];
+                const Float r = right_rows[i];
+                failed |= l != match_map[r];
+            }
+            REQUIRE(!failed);
+        }
+    }
+
+    void check_nans(const kmeans::train_result<>& result) {
+        const auto [centroids, labels, iteration_count] = unpack_result(result);
+
+        SECTION("there is no NaN in centroids") {
+            REQUIRE(te::has_no_nans(centroids));
+        }
+    }
+
+    void check_nans(const kmeans::infer_result<>& result) {
+        const auto [labels, objective_function] = unpack_result(result);
+
+        SECTION("there is no NaN in objective function values") {
+            REQUIRE(te::has_no_nans(labels));
+        }
+    }
+
+private:
+    static auto unpack_result(const kmeans::train_result<>& result) {
+        const auto centroids = result.get_model().get_centroids();
+        const auto labels = result.get_labels();
+        const auto iteration_count = result.get_iteration_count();
+        return std::make_tuple(centroids, labels, iteration_count);
+    }
+    static auto unpack_result(const kmeans::infer_result<>& result) {
+        const auto labels = result.get_labels();
+        const auto objective_function = result.get_objective_function_value();
+        return std::make_tuple(labels, objective_function);
+    }
+};
+
+using kmeans_types = COMBINE_TYPES((float, double), (kmeans::method::lloyd_dense));
+
+TEMPLATE_LIST_TEST_M(kmeans_batch_test,
+                     "kmeans degenerated test",
+                     "[kmeans][batch]",
+                     kmeans_types) {
+    using oneapi::dal::detail::empty_delete;
+    using Float = std::tuple_element_t<0, TestType>;
+    Float data[] = { 0.0, 5.0, 0.0, 0.0, 0.0, 1.0, 1.0, 4.0, 0.0, 0.0, 1.0, 0.0, 0.0, 5.0, 1.0 };
+    homogen_table x{ data, 3, 5, empty_delete<const Float>() };
+    Float labels[] = { 0, 1, 2 };
+    homogen_table y{ labels, 3, 1, empty_delete<const Float>() };
+    this->exact_checks(x, x, x, y, 3, 2, 0.0, 0.0, false);
+}
+
+TEMPLATE_LIST_TEST_M(kmeans_batch_test, "kmeans relocation test", "[kmeans][batch]", kmeans_types) {
+    using oneapi::dal::detail::empty_delete;
+    using Float = std::tuple_element_t<0, TestType>;
+
+    Float data[] = { 0, 0, 0.5, 0, 0.5, 1, 1, 1 };
+    homogen_table x{ data, 4, 2, empty_delete<const Float>() };
```

Comment:
```suggestion
    const auto x = homogen_table::wrap(data, 4, 2);
```

Please apply to other cases.